src/peer_digest.cc

```diff
@@ -531,13 +521,12 @@ peerDigestSwapInCBlock(void *data, char *buf, ssize_t size)
 
     assert(fetch->state == DIGEST_READ_CBLOCK);
 
-    if (peerDigestFetchedEnough(fetch, buf, size, "peerDigestSwapInCBlock"))
+    if (peerDigestFetchedEnough(fetch, size, "peerDigestSwapInCBlock"))
         return -1;
 
     if (size >= (ssize_t)StoreDigestCBlockSize) {
-        PeerDigest *pd = fetch->pd;
+        const auto pd = fetch->pd.get();
 
-        assert(pd);
```

Comment:
If it is accurate, let's keep this assertion. We do dereference `pd` later in this code.